app/backend/src/couchers/servicers/account.py

```diff
@@ -37,6 +37,18 @@ def _check_password(user, field_name, request, context):
         context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.NO_PASSWORD)
 
 
+def _abort_if_terrible_password(field_name, request, context):
+    """
+    Internal utility function: given a request with a StringValue `field_name` field, aborts if password is unforgivably insecure
+    """
+    if len(getattr(request, field_name).value) < 8:
+        context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.PASSWORD_TOO_SHORT)
+
+    # check for most common weak passwords (not meant to be an exhaustive check!)
+    if getattr(request, field_name).value.lower() in ("password", "12345678", "couchers", "couchers1"):
+        context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.INSECURE_PASSWORD)
```

Comment:
This is a good idea to keep the password complexity check isolated to a different function, for maintainability (though I don't foresee this being used anywhere else). Though maybe `_check_password_complexity` might be a more descriptive name :P.

It's probably easier to just pass in the password and context for this function. (The `getattr` mess above was because the password field has a different name in `ChangePassword` and `ChangeEmail`, and it might not be in the proto, so we couldn't pass it straight in. That could maybe be refactored to work on password or None too though).

Reply:
> It's probably easier to just pass in the password and context for this function. (The getattr mess above was because the password field has a different name in ChangePassword and ChangeEmail, and it might not be in the proto, so we couldn't pass it straight in. That could maybe be refactored to work on password or None too though).

Ah, okay. I refactored it with 558bcdc